manifests/init.pp

```diff
@@ -1,135 +1,136 @@
-  $libsudo_vas_location      = 'USE_DEFAULTS',
-  $always_query_group_plugin = 'USE_DEFAULTS',
-  $hiera_merge_sudoers       = false,
+  String[1]                                      $package                   = 'sudo',
+  Optional[String[1]]                            $package_source            = undef,
+  Enum['present','absent']                       $package_ensure            = 'present',
+  Boolean                                        $package_manage            = true,
+  Optional[Stdlib::Absolutepath]                 $package_adminfile         = undef,
+  Stdlib::Absolutepath                           $config_dir                = '/etc/sudoers.d',
+  String[1]                                      $config_dir_group          = 'root',
+  Stdlib::Filemode                               $config_dir_mode           = '0750',
+  Enum['present', 'absent', 'directory', 'link'] $config_dir_ensure         = 'directory',
+  Boolean                                        $config_dir_purge          = true,
+  Hash                                           $sudoers                   = {},
+  Boolean                                        $sudoers_manage            = true,
+  Stdlib::Absolutepath                           $config_file               = '/etc/sudoers',
+  String[1]                                      $config_file_group         = 'root',
+  String[1]                                      $config_file_owner         = 'root',
+  Stdlib::Filemode                               $config_file_mode          = '0440',
+  Boolean                                        $requiretty                = true,
+  Boolean                                        $visiblepw                 = false,
+  Boolean                                        $always_set_home           = true,
+  Boolean                                        $envreset                  = true,
+  Array                                          $envkeep                   = ['COLORS','DISPLAY','HOSTNAME','HISTSIZE','INPUTRC','KDEDIR','LS_COLORS','MAIL','PS1','PS2','QTDIR','USERNAME','LANG','LC_ADDRESS','LC_CTYPE','LC_COLLATE','LC_IDENTIFICATION','LC_MEASUREMENT','LC_MESSAGES','LC_MONETARY','LC_NAME','LC_NUMERIC','LC_PAPER','LC_TELEPHONE','LC_TIME','LC_ALL','LANGUAGE','LINGUAS','_XKB_CHARSET','XAUTHORITY'], #lint:ignore:140chars
+  String[1]                                      $secure_path               = '/sbin:/bin:/usr/sbin:/usr/bin',
+  Boolean                                        $root_allow_all            = true,
+  Boolean                                        $includedir                = true,
+  Boolean                                        $include_libsudo_vas       = false,
+  Stdlib::Absolutepath                           $libsudo_vas_location      = '/opt/quest/lib64/libsudo_vas.so',
+  Optional[Boolean]                              $always_query_group_plugin = undef,
+  Boolean                                        $hiera_merge_sudoers       = false,
```

Comment:
Should we opt to do as we have done in other modules and scrap "merge" parameters and instead rely on native merge in Hiera itself? Might be suitable to keep it in its own commit (or maybe pull request though)?

Reply:
Yes sure, removing hiera_merge will ease testing quite a lot :)
Let me remove $hiera_merge_sudoers and the according tests too.

Reply:
removed with commit [f86c8b5](https://github.com/Ericsson/puppet-module-sudo/pull/39/commits/f86c8b5c6fad97cdcf24c524e93edd791ce2a278)